lib/SILOptimizer/Analysis/ARCAnalysis.cpp

```diff
@@ -1154,3 +1156,132 @@ SILInstruction *swift::findReleaseToMatchUnsafeGuaranteedValue(
   }
   return nullptr;
 }
+
+void EpilogueARCContext::initializeDataflow() {
+  // Find the exit blocks.
+  auto Throw = F->findThrowBB();
+  auto Return = F->findReturnBB();
+  if (Throw != F->end())
+    ExitBlocks.insert(&*Throw);
+  if (Return != F->end())
+    ExitBlocks.insert(&*Return);
+  // Return and throw blocks BBSetOut is initialized to true.
+  for (auto &B : *F) {
+    EpilogueARCBlockStates[&B] =
+              new (BPA.Allocate()) EpilogueARCBlockState();
+  }
+}
+
+void EpilogueARCContext::convergeDataflow() {
+  // Keep iterating until Changed is false.
+  bool Changed = false;
+  do {
+    Changed = false;
+    // Iterate until the data flow converges.
+    for (SILBasicBlock *B : PO->getPostOrder()) {
+      auto BS = EpilogueARCBlockStates[B];
+      // Merge in all the successors.
+      if (!B->succ_empty()) {
+        auto Iter = B->succ_begin();
+        BS->BBSetOut = EpilogueARCBlockStates[*Iter]->BBSetIn;
+        Iter = std::next(Iter);
+        for (auto E = B->succ_end(); Iter != E; ++Iter) {
+	   BS->BBSetOut &= EpilogueARCBlockStates[*Iter]->BBSetIn;
+        }
+      } else if (isExitBlock(B)) {
+        // We set the BBSetOut for exit blocks.
+        BS->BBSetOut = true;
+      }
+
+      // If an epilogue ARC instruction or blocking operating has been identified
+      // then there is no point visiting every instruction in this block.
+      if (!BS->BBSetOut) {
+        Changed |= (BS->BBSetIn != BS->BBSetOut);
+        BS->BBSetIn = BS->BBSetOut;
+        continue;
```

Comment:
Actually the condition whether to iterate over all instructions should be:
if (BSSetOut && IsFirstIteration)
because once you iterated over the instructions BSSetIn becomes false and you don't have to iterate over the instructions in the next dataflow-iteration again.